libass/ass_font.c

```diff
@@ -457,8 +458,16 @@ static int add_face(ASS_FontSelector *fontsel, ASS_Font *font, uint32_t ch)
     set_font_metrics(face);
 
     font->faces[font->n_faces] = face;
-    font->faces_uid[font->n_faces++] = uid;
-    return font->n_faces - 1;
+    font->faces_uid[font->n_faces] = uid;
+    if (!ass_create_hb_font(font, font->n_faces)) {
+        FT_Done_Face(font->faces[font->n_faces]);
```

Comment:
Nit: I feel this would be clearer as `FT_Done_Face(face);`

Reply:
Sure; done.